specification.md

```diff
@@ -1998,19 +2014,18 @@ The workflow id defined cannot be the same id of the workflow there the branch i
 Exceptions can occur during execution of Parallel state branches.
 
 By default, exceptions that are not handled within branches stop branch execution and are propagated 
-to the Parallel state.
+to the Parallel state and should be handled with its `onErrors` definition.
 
-If the parallel states branch defines actions, all exceptions are propagated to the parallel state 
-and can be handled with the parallel states `onError` definition.
+If the parallel states branch defines actions, all exceptions that arise from executing these actions
+ are propagated to the parallel state 
+and can be handled with the parallel states `onErrors` definition.
```

Comment:
I think we should call out that the very first error to be caught should be used to determine control-flow for the workflow execution, that the workflow runtime should not be expected to gather and somehow act upon all exceptions thrown by all parallel branches.

It might be left unsaid perhaps, but I think this leaves runtime implementations to pro-actively cancel and/or ignore parallel branches once the first error bubbles out which could have dire consequences for workflow execution correctness (imagine the case where actions a & b are executed in parallel.  Action a fails with an error that should result in a retry of the current state, if b is still executing when the retry is attempted, then this could cause serious harm having both old and new instances of action b executing at the same time.  I'm not sure if this level of detail belongs in the spec here, or is left to the runtimes to ensure correctness of their workflow executions.  Thoughts?

Reply:
ok i think that was implied but putting this info in here would make sense. will update